scripts/combine/setupCombineWMass.py

```diff
@@ -147,12 +164,19 @@ def main(args,xnorm=False):
     if args.sumChannels or xnorm or name in ["ZMassDilepton"]:
         cardTool.setChannels(["inclusive"])
         cardTool.setWriteByCharge(False)
+    else:
+        cardTool.setChannels(args.recoCharge)
+        if args.forceRecoChargeAsGen:
+            cardTool.setExcludeProcessForChannel("plus", ".*qGen0")
+            cardTool.setExcludeProcessForChannel("minus", ".*qGen1")
+            
     if xnorm:
         histName = "xnorm"
         cardTool.setWriteByCharge(False)
         cardTool.setHistName(histName)
         cardTool.setNominalName(histName)
-        datagroups.select_xnorm_groups() # only keep processes where xnorm is defined
+        ### FIXME: this line was repeated (appears above), I am not yet sure it must be used twice
```

Comment:
I think you were adding the duplication and don't think we need to have it twice. I think the lines from above can go here then we have one "if xnorm:" statement less

Reply:
Ok, at some points I found duplicates and it could be they were removed and added backs in different merges, I'll try to simplify that part